components/plugin_marketplace/marketplace_modal.tsx

```diff
@@ -84,81 +90,89 @@ export const InstalledPlugins = ({installedPlugins, changeTab}) => {
     return <MarketplaceList plugins={installedPlugins}/>;
 };
 
-InstalledPlugins.propTypes = {
-    installedPlugins: PropTypes.array.isRequired,
-    changeTab: PropTypes.func,
+export type MarketplaceModalProps = {
+    show: boolean;
+    plugins: MarketplacePlugin[];
+    installedPlugins: MarketplacePlugin[];
+    siteURL: string;
+    pluginStatuses?: Dictionary<PluginStatusRedux>,
+    actions: {
+        closeModal: () => void;
+        fetchPlugins: (localOnly?: boolean) => Promise<{error?: Error}>;
+        filterPlugins(filter: string): Promise<{error?: Error}>;
+    };
+};
+
+type MarketplaceModalState = {
+    tabKey: any;
+    loading: boolean;
+    serverError: Error | undefined;
+    filter: string;
 };
 
 // MarketplaceModal is the plugin marketplace.
-export class MarketplaceModal extends React.PureComponent {
-    static propTypes = {
-        show: PropTypes.bool,
-        plugins: PropTypes.array.isRequired,
-        installedPlugins: PropTypes.array.isRequired,
-        siteURL: PropTypes.string.isRequired,
-        pluginStatuses: PropTypes.object.isRequired,
-        actions: PropTypes.shape({
-            closeModal: PropTypes.func.isRequired,
-            fetchPlugins: PropTypes.func.isRequired,
-            filterPlugins: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+export class MarketplaceModal extends React.PureComponent<MarketplaceModalProps, MarketplaceModalState> {
+    private filterRef: React.RefObject<QuickInput>;
 
-    constructor(props) {
+    constructor(props: MarketplaceModalProps) {
         super(props);
 
         this.state = {
             tabKey: MarketplaceTabs.ALL_PLUGINS,
             loading: true,
-            serverError: null,
+            serverError: undefined,
```

Comment:
You can remove this line, and will still be undefined.

Reply:
I'm 1/5 that it's easier to read, if all state values are initialized here. That way a reader doesn't have to double check what the default value ist.